File: mediator/java/IChatRoomMediator.java
package Mediator;


// Colleague (User) nesneler arasındaki iletişim arayüzünü tanımlar.
// UML diyagramındaki Mediator arayüzüne denk gelmektedir.
interface IChatRoomMediator {
    void sendMessage(String message, int userId);
    void addUserInRoom(User user);
}

File: mediator/kotlin/IChatRoomMediator.kt
package Mediator


// Colleague (User) nesneler arasındaki iletişim arayüzünü tanımlar.
// UML diyagramındaki Mediator arayüzüne denk gelmektedir.
internal interface IChatRoomMediator {
    fun sendMessage(message: String, userId: Int)
    fun addUserInRoom(user: User)
}
